File: Python/BOJ/4000/4949_균형잡힌 세상.py
while True:
    bracket = []
    sentence = input()

    if sentence == ".":
        break

    for i in sentence:
        if i == "(" or i == "[":
            bracket.append(i)
        elif i == ")":
            if bracket and bracket[-1] == "(":
                bracket.pop()
            else:
                bracket.append(")")
        elif i == "]":
            if bracket and bracket[-1] == "[":
                bracket.pop()
            else:
                bracket.append("]")
    if bracket:
        print("no")
    else:
        print("yes")

File: Swift/BOJ/4000/4949_균형잡힌 세상.swift
while true {
  let input = readLine()!

  if input == "." { break }

  var stack: [Character] = []

  for c in input {
    if c == "(" || c == "[" {
      stack.append(c)
    } else if c == ")" {
      if stack.last == "(" {
        stack.popLast()
      } else {
        stack.append(c)
        break
      }
    } else if c == "]" {
      if stack.last == "[" {
        stack.popLast()
      } else {
        stack.append(c)
        break
      }
    }
  }
  print(stack.count == 0 ? "yes" : "no")
}
